common/src/main/java/com/wynntils/overlays/StatusEffectsOverlay.java

```diff
@@ -108,4 +136,74 @@ private void updateTextRenderSetting() {
     protected TextRenderSetting getTextRenderSetting() {
         return textRenderSetting;
     }
+
+    private Stream<RenderedStatusEffect> stackEffects(List<StatusEffect> effects) {
+        Map<String, RenderedStatusEffect> effectsToRender = new LinkedHashMap<>();
+
+        for (StatusEffect effect : effects) {
+            RenderedStatusEffect entry = effectsToRender.get(effect.asString().getString());
+            if (entry == null) {
+                entry = new RenderedStatusEffect(effect);
+                effectsToRender.put(effect.asString().getString(), entry);
+            }
+            entry.setCount(entry.getCount() + 1);
+        }
+        return effectsToRender.values().stream();
+    }
+
+    private static final class RenderedStatusEffect {
+        private int count = 0;
+        private final StatusEffect effect;
+
+        private RenderedStatusEffect(StatusEffect effect) {
+            this.effect = effect;
+        }
+
+        private StyledText getRenderedText() {
+            if (this.count == 1) {
+                return this.effect.asString();
+            }
+            String modifierString = this.effect.getModifier().getString();
+            StyledText modifierText;
+
+            // look for either a - or a +
+            int minusIndex = modifierString.indexOf('-');
+            int plusIndex = modifierString.indexOf('+');
+            int index = max(minusIndex, plusIndex);
+
+            // We can simply put the count string at the start
+            if (index == -1) {
+                modifierText = StyledText.fromString(ChatFormatting.GRAY + (this.count + "x"))
+                        .append(this.effect.getModifier());
```

Comment:
```suggestion
            if (index == -1) {
                 // We can simply put the count string at the start
                modifierText = StyledText.fromString(ChatFormatting.GRAY + (this.count + "x"))
                        .append(this.effect.getModifier());
```

Reply:
Should I just commit all the changes you proposed?

Reply:
yeah